code_generation/quantity.py

```diff
@@ -129,3 +219,15 @@ def reserve_scope(self, scope):
             )
         )
         raise Exception
+
+    # if the shifted version of a quantity already exists in the input,
+    # this function can be used to register an
+    # branch from the input as a shifted version of a quantity
+    def register_external_shift(self, shift, scope, external_name):
+        if scope not in self.shifts.keys():
+            self.shifts[scope] = {}
+        if scope == "global":
+            for allscope in self.shifts.keys():
+                self.shifts[allscope][shift] = external_name
+        else:
+            self.shifts[scope][shift] = external_name
```

Comment:
I'd put everything into global and remove the method argument `scope`. There is no need for different scopes in nanoAOD quantities

Reply:
Furthermore, you need to take care of shifting the children via the usual shift-function

Reply:
I tried this, but i think this does not work.. 

When the writecall function is run, `get_leaf(shift, scope)` is run. Since the NanoAODQuantity only exists in global, the `BuildMetVector` function, which currently sits in the `mt` scope does not get the shifted version

Reply:
I'd assume that this is because the shift was not correctly propagated to the children (second point above). Check how it is done in the shift function ( https://github.com/KIT-CMS/CROWN/blob/main/code_generation/quantity.py#L52-L55 ). If the scope is global, children within all scopes must be shifted with the respective scope.

Reply:
For the children, it does work, only for the variable itself, it does not

Reply:
quantity class changes are reversed